ferrilata-devilline/Controllers/BadgesController.cs

```diff
@@ -1,29 +1,57 @@
-﻿using ferrilata_devilline.Services.Interfaces;
-using Microsoft.AspNetCore.Mvc;
+﻿using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json.Linq;
+using System.Collections.Generic;
+using ferrilata_devilline.Models.DTOs;
+using ferrilata_devilline.Services;
+using ferrilata_devilline.Services.Helpers;
+using ferrilata_devilline.Services.Helpers.ObjectTypeCheckers;
+using ferrilata_devilline.Services.Interfaces;
 
 namespace ferrilata_devilline.Controllers
 {
     public class BadgesController : Controller
     {
-        private readonly IBadgeService _badgeService;
+        JTokenAnalyzer _jTokenAnalyzer;
+        private readonly IBadgeAndLevelService _badgeService;
 
-        public BadgesController(IBadgeService badgeService)
+        public BadgesController(JsonSchemaService SchemaService, IBadgeAndLevelService badgeService)
         {
+            _jTokenAnalyzer = new JTokenAnalyzer(SchemaService);
```

Comment:
i recall that we said that we'll merge as is, and then change later, but we can totally clarify this tomorrow. At least we have tests now, so it will be easy to see if the less convoluted solution works.